core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -294,6 +322,20 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     }
   }
 
+  test("add jar when path contains spaces") {
+    withTempDir { dir =>
+      val sep = File.separator
+      val tmpDir = Utils.createTempDir(dir.getAbsolutePath + sep + "test space")
+      val tmpJar = File.createTempFile("test", ".jar", tmpDir)
```

Comment:
can we also test if the jar file name contains space?

Reply:
ok , i will check and update you

Reply:
1. addJar is working fine even if jar file name contain space.
2. addFile is not working , need to correct it.

Reply:
If it's hard to fix list jars, we can do it in another PR

Reply:
ok, i will raise one jira for this and will raise separate.

Reply:
ok i will raise jira for list jars and files ,will create separate PR.

Reply:
What is the issue with listJars() output there?

Reply:
listJars() and listFiles() function will show file name in encoded form.
Please check this above image :

**suppose : filename "ankit raj boudh.jar" or "ankit raj boudh.json"
then : listJars() output : ankit%20raj%20boudh.jar
         : listFiles() output : ankit%20raj%20boudh.json**

Reply:
listJars() and listFiles() function will show file name in encoded form when file path contains spaces.
Please check this above image :

suppose : filename "ankit raj boudh.jar" or "ankit raj boudh.json"
then : listJars() output : ankit%20raj%20boudh.jar
: listFiles() output : ankit%20raj%20boudh.json

Reply:
But that's not what it shows - it shows a URI, which does need escaping. Is it a change of behavior to show the URI vs path here?

Reply:
Change is not for lisrJars() and listFiles() behaviour, this is only for addJar and addFile function

Reply:
Yes, but I'm asking what behavior changes in listJars that you are suggesting is a problem. If before this URI had a space in it, it was wrong. If it was a simple path as string before, then yes that's a behavior change we need to avoid, I think.

Reply:
I am suggesting showing path will be good for listJars and listFile, instead of showing URI.
because it will not be readable.

Reply:
I'm not sure if you're saying there is no change with this PR, or there is a change but you think the new output is less desirable? what's wrong with a correct URI in general? but, if it was displaying paths before, then, stick with paths.

Reply:
i think there is confusion , let me clear

Reply:
1. Previous addFile and listFile function behaviour : 
```
addFile : sc.addFile("/Users/testdir/test3.conf") 
listFile : sc.listFile 
output : res1: Seq[String] = Vector(file:/Users/testdir/test3.conf)
```

2. After this PR addFile and listFile function behaviour :
```
addFile : sc.addFile("/Users/test dir/test3.conf") - here path contain space 
listFile : sc.listFile 
output : res1: Seq[String] = Vector(file:/Users/test%20dir/test3.conf) - output will look like this (this is previous behaviour)
```

here listFiles function behaviour not change.

Reply:
same thing for addJar and listJars

1. Previous addJar and listJars function behaviour :

```
addJar : sc.addJar("/Users/testdir/test3.jar") 
listJars : sc.listJars
output : res1: Seq[String] = Vector(file:/Users/testdir/test3.jar)
```

2. After this PR addFile and listFile function behaviour :

```
addJar : sc.addJar("/Users/test dir/test3.jar") - here path contain space 
listJars : sc.listJars 
output : res1: Seq[String] = Vector(file:/Users/test%20dir/test3.jar) - output will look like this (this is previous behaviour)
```

Reply:
what i have suggested output of listJars and listFiles function should be like this when file name contain space :

1.  Previous :
 output : res1: Seq[String] = Vector(file:/Users/**test%20dir**/test3.jar) 

2. My suggestion :
output : res1: Seq[String] = Vector(file:/Users/**test dir**/test3.jar)

Note : this suggestion related change not there in this PR, for this i will raise another PR in this Jira SPARK-30180

Reply:
@srowen , I hope now it's clear for you

Reply:
No, that makes the result invalid. A URI like that can't have a space. The former representation is the correct URI for a file path with a space.

Reply:
showing path like this is ok ?

**output : res1: Seq[String] = Vector(file:/Users/test%20dir/test3.jar)**

Reply:
ok then no need to fix this jira SPARK-30180

Reply:
then please review this PR

Reply:
Yep, let's hold fire on opening more issues and JIRAs before discussing these

Reply:
ok

Reply:
@srowen, Thank you for correction and if any other point please let me know